src/crawlee/beautifulsoup_crawler/_beautifulsoup_crawler.py

```diff
@@ -83,157 +72,18 @@ def __init__(
                 as successful responses.
             kwargs: Additional keyword arguments to pass to the underlying `BasicCrawler`.
         """
-        self._parser = parser
-
-        kwargs['_context_pipeline'] = (
-            ContextPipeline()
-            .compose(self._make_http_request)
-            .compose(self._parse_http_response)
-            .compose(self._handle_blocked_request)
-        )
-
-        kwargs.setdefault(
-            'http_client',
-            HttpxHttpClient(
-                additional_http_error_status_codes=additional_http_error_status_codes,
-                ignore_http_error_status_codes=ignore_http_error_status_codes,
-            ),
-        )
-
-        kwargs.setdefault('_logger', logging.getLogger(__name__))
-
-        super().__init__(**kwargs)
-
-    async def _make_http_request(self, context: BasicCrawlingContext) -> AsyncGenerator[HttpCrawlingContext, None]:
-        """Executes an HTTP request using a configured HTTP client.
-
-        Args:
-            context: The crawling context from the `BasicCrawler`.
-
-        Yields:
-            The enhanced crawling context with the HTTP response.
-        """
-        result = await self._http_client.crawl(
-            request=context.request,
-            session=context.session,
-            proxy_info=context.proxy_info,
-            statistics=self._statistics,
-        )
-
-        yield HttpCrawlingContext(
-            request=context.request,
-            session=context.session,
-            proxy_info=context.proxy_info,
-            add_requests=context.add_requests,
-            send_request=context.send_request,
-            push_data=context.push_data,
-            get_key_value_store=context.get_key_value_store,
-            log=context.log,
-            http_response=result.http_response,
-        )
 
-    async def _handle_blocked_request(
-        self,
-        context: BeautifulSoupCrawlingContext,
-    ) -> AsyncGenerator[BeautifulSoupCrawlingContext, None]:
-        """Try to detect if the request is blocked based on the HTTP status code or the response content.
-
-        Args:
-            context: The current crawling context.
+        async def final_step(
```

Comment:
I personally prefer non-generator functions. On the other hand, I even more strongly prefer to have "only one way of doing things". So this would just add small convenience in more flexible way of calling compose, but it would mean some extra code and maintaining for us. So in the end, I would be (not strongly) against it.